src/agent/coverage/src/block.rs

```diff
@@ -98,6 +101,13 @@ impl ModuleCov {
 #[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct BlockCov {
     /// Offset of the block, relative to the module base load address.
+    //
+    // These offsets are represented as `u64` values, but since they are offsets
+    // within an assumed well-formed module, they should never exceed a `u32`.
+    // We thus assume that they can be losslessly serialized to an `f64`.
+    //
+    // If we need to handle malformed binaries or arbitrary addresses, then this
```

Comment:
Maybe add an assertion in the constructor?

Reply:
I'll be addressing this separately in a follow-up PR, with explicit failure if the assumption is violated.